source/modals/bountypostmodal.js

```diff
@@ -0,0 +1,112 @@
+const { getRankUpdates } = require('../helpers');
+
+const customId = "bountypostmodal";
+module.exports = new InteractionWrapper(customId, 3000,
+	/** Serialize data into a bounty, then announce with showcase embed */
+	async (interaction, [slotNumber, isEvergreen]) => {
+		const title = interaction.fields.getTextInputValue("title");
+		const description = interaction.fields.getTextInputValue("description");
+		const imageURL = interaction.fields.getTextInputValue("imageURL");
+		const startTimestamp = parseInt(interaction.fields.getTextInputValue("startTimestamp"));
+		const endTimestamp = parseInt(interaction.fields.getTextInputValue("endTimestamp"));
+		const shouldMakeEvent = startTimestamp && endTimestamp;
+
+		const rawBounty = {
+			userId: interaction.user.id,
+			guildId: interaction.guildId,
+			slotNumber: parseInt(slotNumber),
+			isEvergreen: isEvergreen === "true",
+			title,
+			description
+		};
+		const errors = [];
+
+		if (imageURL) {
+			try {
+				new URL(imageURL);
+				rawBounty.attachmentURL = imageURL;
+			} catch (error) {
+				errors.push(error.message);
+			}
+		}
+
+		if (shouldMakeEvent) {
+			if (!startTimestamp) {
+				errors.push("Start timestamp must be an integer.");
+			} else if (!endTimestamp) {
+				errors.push("End timestamp must be an integer.");
+			} else {
+				if (startTimestamp > endTimestamp) {
+					errors.push("End timestamp was before start timestamp.");
+				}
+
+				const nowTimestamp = Date.now() / 1000;
+				if (nowTimestamp >= startTimestamp) {
+					errors.push("Start timestamp must be in the future.");
+				}
+
+				if (nowTimestamp >= endTimestamp) {
+					errors.push("End timestamp must be in the future.");
+				}
+
+				if (startTimestamp >= nowTimestamp + (5 * YEAR_IN_MS)) {
+					errors.push("Start timestamp cannot be 5 years in the future or further.");
+				}
+
+				if (endTimestamp >= nowTimestamp + (5 * YEAR_IN_MS)) {
+					errors.push("End timestamp cannot be 5 years in the future or further.");
+				}
+			}
+		}
+
+		if (errors.length > 0) {
+			interaction.reply({ content: `The following errors were encountered while posting your bounty **${title}**:\n• ${errors.join("\n• ")}`, ephemeral: true });
+		} else {
+			const poster = await database.models.Hunter.findOne({ where: { userId: interaction.user.id, guildId: interaction.guildId } });
+			poster.addXP(interaction.guild, 1, true).then(() => {
+				getRankUpdates(interaction.guild);
+			});
+
+			if (shouldMakeEvent) {
+				const eventPayload = {
+					name: `Bounty: ${title}`,
+					description,
+					scheduledStartTime: startTimestamp * 1000,
+					scheduledEndTime: endTimestamp * 1000,
+					privacyLevel: 2,
+					entityType: GuildScheduledEventEntityType.External,
+					entityMetadata: { location: `${interaction.member.displayName}'s #${slotNumber} Bounty` }
+				};
+				if (imageURL) {
+					eventPayload.image = imageURL;
+				}
+				const event = await interaction.guild.scheduledEvents.create(eventPayload);
```

Comment:
Is there a reason to save the promise results for the event here? We don't end up using the returned value to save/attach information to the Bounty itself, so this just...hangs there for garbage collection. It may be wise to at minimum `try..catch` this to tell the user if the event creation had an error that we didn't manage to catch with validation.

Reply:
The `event.id` is used in L88.